hubtools/src/lib.rs

```diff
@@ -796,4 +808,43 @@ impl RawHubrisArchive {
     pub fn replace(&mut self, data: Vec<u8>) {
         self.image.replace(data);
     }
+
+    pub fn verify(
+        &mut self,
+        src_cmpa: PathBuf,
+        src_cfpa: PathBuf,
+    ) -> Result<(), Error> {
+        let cmpa = {
+            let contents = std::fs::read(&src_cmpa)
+                .map_err(|e| Error::FileReadFailed(src_cmpa, e))?;
+
+            if contents.len() != 512 {
+                return Err(Error::BadCMPALen(contents.len()));
+            }
+
+            let mut cmpa_bytes = [0u8; 512];
+            cmpa_bytes.copy_from_slice(&contents[..]);
+            lpc55_areas::CMPAPage::from_bytes(&cmpa_bytes)
+                .map_err(|_| Error::BadCMPA)?
+        };
+
+        let cfpa = {
+            let contents = std::fs::read(&src_cfpa)
+                .map_err(|e| Error::FileReadFailed(src_cfpa, e))?;
+
+            if contents.len() != 512 {
+                return Err(Error::BadCFPALen(contents.len()));
+            }
+
+            let mut cfpa_bytes = [0u8; 512];
+            cfpa_bytes.copy_from_slice(&contents[..]);
+            lpc55_areas::CFPAPage::from_bytes(&cfpa_bytes)
+                .map_err(|_| Error::BadCFPA)?
+        };
+
+        lpc55_sign::verify::log_verify_only_on_failure();
+        lpc55_sign::verify::verify_image(&self.image.data, cmpa, cfpa)
+            .map_err(|e| Error::Lpc55(e))?;
```

Comment:
How much information do we get back from this `e`? Looking at the output from your run:

```
[ERROR lpc55_sign::verify] ROTKH in CMPA does not match RKH table in image
Error: LPC55 support error: verification failed; see log for details

Caused by:
    verification failed; see log for details
```

I think we'd really like the `ROTKH in CMPA does not match RKH table in image` message to be in the error and not just the log (where does that log end up if I'm using this as a library from a running service?), although this looks like an issue in `lpc55_sign` and not this PR.

Reply:
We only get a yes/no answer right now from verify. I was debating switching verify to have proper error type and I think your point about running this as a service is a good argument for cleaning that up.

Reply:
I fixed up the `verify_image` function to return a proper error